src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/LocalDBAPI.cs

```diff
@@ -10,25 +10,41 @@ namespace Microsoft.Data
 {
     internal static partial class LocalDBAPI
     {
-        private const string const_localDbPrefix = @"(localdb)\";
+        private const string LocalDbPrefix = @"(localdb)\";
+        private const string LocalDbPrefix_NP = @"np:\\.\pipe\LOCALDB#";
 
 
         [UnmanagedFunctionPointer(CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         private delegate int LocalDBFormatMessageDelegate(int hrLocalDB, uint dwFlags, uint dwLanguageId, StringBuilder buffer, ref uint buflen);
 
         // check if name is in format (localdb)\<InstanceName - not empty> and return instance name if it is
+        // localDB can also have a format of np:\\.\pipe\LOCALDB#<some number>\tsql\query
         internal static string GetLocalDbInstanceNameFromServerName(string serverName)
         {
-            if (serverName == null)
-                return null;
-            serverName = serverName.TrimStart(); // it can start with spaces if specified in quotes
-            if (!serverName.StartsWith(const_localDbPrefix, StringComparison.OrdinalIgnoreCase))
-                return null;
-            string instanceName = serverName.Substring(const_localDbPrefix.Length).Trim();
-            if (instanceName.Length == 0)
-                return null;
-            else
-                return instanceName;
+            if (serverName is not null)
+            {
+                // it can start with spaces if specified in quotes
+                // Memory allocation is reduced by using ReadOnlySpan
+                ReadOnlySpan<char> input = serverName.AsSpan().TrimStart();
+                bool isLocalDb = input.StartsWith(LocalDbPrefix.AsSpan().Trim(), StringComparison.OrdinalIgnoreCase)
```

Comment:
Why are the constants being trimmed here?

Reply:
fixed.